src/components/CommentsSection.jsx

```diff
@@ -1,26 +1,68 @@
-import { useState, useEffect } from "react";
+import { useState, useEffect, useContext } from "react";
 import Comment from "./Comment";
 import Loading from "./Loading";
-import { fetchComments } from "../../utils/api";
+import { fetchComments, postComment } from "../../utils/api";
+import { UserContext } from "../context/User";
 
 export default function CommentsSection({ articleId }) {
+  const { user, setUser } = useContext(UserContext);
   const [comments, setComments] = useState({});
   const [loading, setLoading] = useState(true);
+  const [newComment, setNewComment] = useState("");
+  const [submitted, setSubmitted] = useState(false);
+  const [error, setError] = useState(null);
 
   useEffect(() => {
     fetchComments(articleId).then((commentsData) => {
       setComments(commentsData);
       setLoading(false);
     });
-  }, []);
+  }, [submitted]);
+
+  const handleChange = ({ target: { value } }) => {
+    setNewComment(value);
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setError(null);
+    postComment(user, newComment, articleId)
+      .then(() => {
+        setNewComment("");
+        setSubmitted(true);
+      })
+      .catch((err) => {
+        console.log(err);
+        setError("Comment unsuccessful, try again.");
+      });
+    setSubmitted(false);
+  };
 
   return loading ? (
     <Loading />
   ) : (
-    <ul>
-      {comments.map((comment) => {
-        return <Comment key={comment.comment_id} comment={comment} />;
-      })}
-    </ul>
+    <>
+      <form onSubmit={handleSubmit}>
+        <label className="flex gap-2">
+          <textarea
+            onChange={handleChange}
+            value={newComment}
+            placeholder="Add a comment"
+            required
+            className="border rounded-xl py-1 px-2 m-auto min-h-[44px] max-h-96 min-w-[390px] max-w-[1000px]"
```

Comment:
start looking into Tailwind responsive sizes. You can set new styles after certain sizes (`lg` is 1024 px). `lg:bg-black` would set the background to black when the screen size is `lg`. 

`max-lg:bg-black` would set the background black until the screen size reaches `lg`. 

Other sizes are xs, sm, lg, xl (can't remember how much higher they go). Check them out! They work with all sorts of styling. You can change `flex` to column on small sizes, row on larger sizes. Images can be `w-full` on small sizes, `w-2/3` on larger sizes, `w-1/2` on even larger sizes. It can be really easy to get the responsiveness like this!